code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MultiIconPushTemplate.kt

```diff
@@ -0,0 +1,112 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.templates
+
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateConstants.DEFAULT_DELETE_ICON_NAME
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.NotificationData
+import com.adobe.marketing.mobile.services.Log
+import org.json.JSONArray
+import org.json.JSONObject
+
+internal class MultiIconPushTemplate(data: NotificationData) : AEPPushTemplate(data) {
+
+    private val SELF_TAG = "MultiIconNotificationTemplate"
+    data class MultiIconTemplateItem(
+        val iconUrl: String,
+        val actionType: PushTemplateConstants.ActionType,
+        val actionUri: String?
+    )
+
+    internal val templateItemList: MutableList<MultiIconTemplateItem>
+    internal var cancelIcon: String? = null
+
+    init {
+        val itemsJson = data.getRequiredString(PushTemplateConstants.PushPayloadKeys.MULTI_ICON_ITEMS)
+        templateItemList = getTemplateItemList(itemsJson)
+            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.MULTI_ICON_ITEMS}\" is invalid.")
+        if (templateItemList.size < 3 || templateItemList.size > 5) {
```

Comment:
lets add 3 as the minimum image count and 5 as the maximum image count to the constants

Reply:
Done, added to default value for min and max image count.